webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/index.tsx

```diff
@@ -1,18 +1,32 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {useState} from 'react';
+import React, {useEffect, useState} from 'react';
 
+import type {DesktopNotificationPermission} from 'components/common/hooks/use_desktop_notification_permission';
+import {getDesktopAppNotificationPermission} from 'components/common/hooks/use_desktop_notification_permission';
 import NotificationPermissionDeniedNotice from 'components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_denied_section_notice';
 import NotificationPermissionNeverGrantedNotice from 'components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_never_granted_section_notice';
 import NotificationPermissionUnsupportedSectionNotice from 'components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_unsupported_section_notice';
 
 import {getNotificationPermission, isNotificationAPISupported, NotificationPermissionDenied, NotificationPermissionNeverGranted} from 'utils/notifications';
 
+import NotificationPermissionDesktopDeniedSectionNotice from './notification_permission_desktop_denied_section_notice';
+
 export default function NotificationPermissionSectionNotice() {
     const isNotificationSupported = isNotificationAPISupported();
 
     const [notificationPermission, setNotificationPermission] = useState(getNotificationPermission());
+    const [desktopNotificationPermission, setDesktopNotificationPermission] = useState<DesktopNotificationPermission>(undefined);
```

Comment:
The code in both this component and `NotificationPermissionTitleTag` seems like it's the same. Could you make a hook that wraps this logic and calls `getDesktopAppNotificationPermission` on first mount to avoid repeating that? That'll make the logic more self-contained.

You could also use that hook in the `NotificationPermissionBar` even though it doesn't care about the result. That way, you also wouldn't have to export `getDesktopAppNotificationPermission`

Reply:
converted to hook